### DISCUSSION TITLE
Excluding domains from MULTI_URIBL

### DISCUSSION THREAD

**Discussion by kgaughan** [rspamd/rspamd#4196] [Q&A]
Short of disabling `MULTI_URIBL` entirely, is there a way to skip certain domains from being checked against it?

Here's the background: In spite of my mailserver not getting an awful lot of traffic and running a copy of unbound locally to cache DNS query results, I've noticed this in my logs:
```
monitored; rspamd_monitored_dns_cb: DNS query blocked on multi.uribl.com (127.0.0.1 returned), possibly due to high volume
```
To try to help the situation, I'd like to skip certain domains, as [recommended on the site](http://uribl.com/refused.shtml). I've poked around [the documentation](https://rspamd.com/doc/modules/rbl.html), but there's no obvious way to do this. There's an undocumented parameter called `whitelist` I noticed in `lualib/plugins/rbl.lua` that might do what I want, and also the documented parameter `whitelist_exception`, but the description doesn't entirely make sense to me, and my knowledge of Lua and rspamd's internals doesn't extend far enough to me figuring it out.

**Comment by kartoffelheinz** [comment]
We're also running into this limit from time to time and would love a way to skip querying RBLs for some domains. I do not think the whitelist parameters could achieve this, we might need a skiplist. Anybody an idea if there is already something implemented but nod documented so far?

**Comment by fatalbanana** [comment]
There is `url_whitelist` setting featuring in default `rbl` module configuration; it's carry-over from `surbl` module & unfortunately not well-documented in its contemporary context.

~~~
  url_whitelist = [
    "https://maps.rspamd.com/rspamd/surbl-whitelist.inc.zst",
    "$LOCAL_CONFDIR/local.d/maps.d/surbl-whitelist.inc.local",
    "${DBDIR}/surbl-whitelist.inc.local",
    "fallback+file://${CONFDIR}/maps.d/surbl-whitelist.inc"
  ];
~~~

You could add domains to `/var/lib/rspamd/surbl-whitelist.inc.local` for example.
